### DISCUSSION TITLE
get only components customized id.

### DISCUSSION THREAD

**Discussion by ronaldohoch** [GrapesJS/grapesjs#4419] [Q&A] [answered]
Hello @artf and other contributors.

I know if i run `editor.DomComponents.componentsById` it brings all the components id in canvas.
![image](https://user-images.githubusercontent.com/2287371/176773546-d55a0fb1-60f6-4fb0-bb73-6f8bf8bf8e3b.png)
But, you can check above, i have an id called MyIdentification and the other autogenerated  ids starting with "i" as we can confirm here: https://github.com/artf/grapesjs/blob/dev/src/code_manager/model/HtmlGenerator.js#L24

In my new feature, i need to facilitate for the user to set an anchor pointing to an id in the document...
But we have these autogenerated ids in the list.

I would like to know if there is a way to get just the custom ids from.

I could filter by id's starting with "i" but we have a lot of words starting with "i" both in English and Portuguese

**Comment by artf** [comment, marked as answer]
Hi @ronaldohoch currently one way to see if the id was manually updated is to check it inside the component attributes
```js
if (component.get('attributes').id) {
 // id updated
}
```

**Reply by ronaldohoch** [reply]
Thank you @artf i'm doing this way, i don't know why i didn't update the discussion.

But, when i update any other trait in the component, it updates the "attributes" with autogenerated id too.
